### ISSUE TITLE
[sanity checker] check for particles of unknown type in the I3MCTree (Trac #884)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#878]
ClSim says:

```text
RuntimeError: It seems you either ran MMC with the "-recc" option or I3MuonSlicer has already been applied.
```

Test file attached. This is 2012 nugen, so the gcd file is:
`/data/sim/sim-new/downloads/GCD/GeoCalibDetectorStatus_2012.56063_V1.i3.gz`

Refer to comment:4 for instructions.

Migrated from https://code.icecube.wisc.edu/ticket/884
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:45", 
    "description": "ClSim says:\n\n{{{\nRuntimeError: It seems you either ran MMC with the \"-recc\" option or I3MuonSlicer has already been applied.\n}}}\n\nTest file attached. This is 2012 nugen, so the gcd file is:\n`/data/sim/sim-new/downloads/GCD/GeoCalibDetectorStatus_2012.56063_V1.i3.gz`\n\nRefer to comment:4 for instructions.", 
    "reporter": "david.schultz", 
    "cc": "david.schultz, claudio.kopper", 
    "resolution": "fixed", 
    "_ts": "1550067105393059", 
    "component": "combo simulation", 
    "summary": "[sanity checker] check for particles of unknown type in the I3MCTree", 
    "priority": "major", 
    "keywords": "clsim", 
    "time": "2015-02-24T05:23:11", 
    "milestone": "", 
    "owner": "olivas", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150224T05:23:34: david.schultz changed attachment from "" to "test.i3"

**Comment by icecube-trac** [comment]
Trac update at 20150519T21:18:54: claudio.kopper changed milestone from "" to "IceSim 4.2"

**Comment by icecube-trac** [comment]
Trac update at 20150916T17:41:09: 
* claudio.kopper commented:

> The MCTree in this test event is broken - it contains an "unknown" particle (at least it is visible as one in the trunk) - this might be related to the new MCTree implementation, but this is not a bug in clsim.
> 
> Assigning this to david.schultz
> 
> Here's a dump of the tree as visible in trunk r137628:
> 
> 
> ```text
> 44 NuMuBar (4.68121e+06m ,-4.07158e+06m ,-4.90997e+06m) (128.36deg ,318.985deg) 0ns 125.34GeV 7.91251e+06m 
>   45 MuPlus (-213.146m ,126.835m ,549.544m) (128.958deg ,318.345deg) 2.63933e+07ns 116.281GeV 360.232m 
>     48 DeltaE (-226.002m ,138.272m ,563.457m) (128.958deg ,318.345deg) 2.63934e+07ns 0.582697GeV 0m 
>     49 DeltaE (-227.019m ,139.176m ,564.558m) (128.958deg ,318.345deg) 2.63934e+07ns 1.01576GeV 0m 
>     50 DeltaE (-288.666m ,194.015m ,631.272m) (128.958deg ,318.345deg) 2.63937e+07ns 0.547283GeV 0m 
>     51 DeltaE (-297.123m ,201.538m ,640.424m) (128.958deg ,318.345deg) 2.63938e+07ns 0.635205GeV 0m 
>     52 PairProd (-348.459m ,247.204m ,695.978m) (128.958deg ,318.345deg) 2.63941e+07ns 0.755961GeV 0m 
>     53 DeltaE (-351.909m ,250.272m ,699.711m) (128.958deg ,318.345deg) 2.63941e+07ns 0.544377GeV 0m 
>     54 DeltaE (-354.165m ,252.279m ,702.153m) (128.958deg ,318.345deg) 2.63941e+07ns 0.81707GeV 0m 
>     55 Brems (-357.405m ,255.162m ,705.66m) (128.958deg ,318.345deg) 2.63941e+07ns 0.775377GeV 0m 
>     25 unknown (-391.667m ,285.64m ,742.738m) (128.958deg ,318.345deg) 2.63943e+07ns 29.6761GeV 0m 
>       56 EMinus (-391.667m ,285.64m ,742.738m) (128.958deg ,318.345deg) 2.63943e+07ns 16.3417GeV 0m 
>     57 DeltaE (-397.059m ,290.436m ,748.572m) (128.958deg ,318.345deg) 2.63943e+07ns 0.701542GeV 0m 
>     58 EPlus (-422.44m ,313.014m ,776.04m) (128.958deg ,318.345deg) 2.63979e+07ns 0.0506501GeV 0m 
>   26 unknown (-213.146m ,126.835m ,549.544m) (128.36deg ,318.985deg) 2.63933e+07ns 9.05853GeV 0m 
>     46 EMinus (-213.146m ,126.835m ,549.544m) (128.36deg ,318.985deg) 2.63933e+07ns 5.28812GeV 0m 
> ```


* claudio.kopper changed priority from "blocker" to "major"
* claudio.kopper changed owner from "claudio.kopper" to "david.schultz"
* claudio.kopper changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20150922T01:40:00: olivas changed milestone from "IceSim 4.2" to "IceSim 5.0"

**Comment by icecube-trac** [comment]
Trac update at 20151007T14:17:10: olivas commented:

> Maybe the best we can do at this point is to add a sanity checker for particles of unknown type?

**Comment by icecube-trac** [comment]
Trac update at 20151207T15:53:14: 
* david.schultz changed cc from "olivas" to "david.schultz, claudio.kopper"
* david.schultz changed summary from "clsim - muon slicer error" to "[sanity checker] check for particles of unknown type in the I3MCTree"
* david.schultz changed description from:

> RuntimeError: It seems you either ran MMC with the "-recc" option or I3MuonSlicer has already been applied.
> 
> Test file attached. This is 2012 nugen, so the gcd file is:
> `/data/sim/sim-new/downloads/GCD/GeoCalibDetectorStatus_2012.56063_V1.i3.gz`

to:

> ClSim says:
> 
> {{{
> RuntimeError: It seems you either ran MMC with the "-recc" option or I3MuonSlicer has already been applied.
> }}}
> 
> Test file attached. This is 2012 nugen, so the gcd file is:
> `/data/sim/sim-new/downloads/GCD/GeoCalibDetectorStatus_2012.56063_V1.i3.gz`
> 
> Refer to comment:4 for instructions.


* david.schultz changed owner from "david.schultz" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20160119T03:34:13: 
* olivas commented:

> In changeset 141004:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="141004"
> added a check for unknown particle type. closes #878.
> ```


* olivas changed status from "assigned" to "closed"
* olivas changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:45: 
* olivas changed milestone from "IceSim 5.0" to ""
* olivas commented:

> Milestone IceSim 5.0 deleted
